scipy/stats/_continuous_distns.py

```diff
@@ -8854,6 +8854,139 @@ def _updated_ctor_param(self):
+
+    .. [1] "Studentized range distribution",
+           https://en.wikipedia.org/wiki/Studentized_range_distribution
+    .. [2] Batista, Ben Dêivide, et al. “Externally Studentized Normal Midrange
+           Distribution.” Ciência e Agrotecnologia, vol. 41, no. 4, 2017, pp.
+           378–389., doi:10.1590/1413-70542017414047716.
+    .. [3] Harter, H. Leon. “Tables of Range and Studentized Range.” The Annals
+           of Mathematical Statistics, vol. 31, no. 4, 1960, pp. 1122–1147.
+           JSTOR, www.jstor.org/stable/2237810. Accessed 18 Feb. 2021.
+    .. [4] Lund, R. E., and J. R. Lund. “Algorithm AS 190: Probabilities and
+           Upper Quantiles for the Studentized Range.” Journal of the Royal
+           Statistical Society. Series C (Applied Statistics), vol. 32, no. 2,
+           1983, pp. 204–210. JSTOR, www.jstor.org/stable/2347300. Accessed 18
+           Feb. 2021.
+
+
+    %(example)s
+
+    """
+    _epsabs = 1e-9  # Allows setting of quad epsabs for class integration fns.
+    _force_fn = None  # "asymptotic", None, or "regular"
+
+    def _argcheck(self, k, v):
+        """Verify args"""
+        return np.all(k > 1) and np.all(v > 0)
+
+    def _munp(self, K, k, v):
+        _a, _b = self._get_support()
+        K = np.atleast_1d(K)
+
+        def _single_moment(K, k, v):
+            user_data = np.array([K, k, v], float).ctypes.data_as(
+                ctypes.c_void_p)
+
+            llc = LowLevelCallable.from_cython(
+                _stats,
+                '_genstudentized_range_moment',
+                user_data
+            )
+
+            return integrate.nquad(
+                llc,
+                ranges=[(-np.inf, np.inf), (0, np.inf), (_a, _b)]
+            )[0]
+
+        return _single_moment(K, k, v)
+
+    def _pdf(self, x, k, v):
+        _a, _b = self._get_support()
+        x = np.atleast_1d(x)
+
+        def _single_pdf(q, k, v):
+            user_data = np.array([q, k, v], float).ctypes.data_as(
+                ctypes.c_void_p)
+
+            llc = LowLevelCallable.from_cython(
+                _stats,
+                '_genstudentized_range_pdf',
+                user_data
+            )
+
+            return integrate.nquad(
+                llc,
+                ranges=[(-np.inf, np.inf), (_a, _b)],
+                opts=dict(epsabs=self._epsabs)
+            )[0]
+
+        ufunc = np.frompyfunc(_single_pdf, 3, 1)
+        return ufunc(x, k, v).astype("float64")
+
+    def _cdf(self, x, k, v):
+        _a, _b = self._get_support()
+        x = np.atleast_1d(x)
+
+        """The cdf"""
+        def _single_cdf(q, k, v):
+            # "When the degrees of freedom V are infinite the probability
+            # integral takes [a] simpler form," and a single asymptotic
+            # integral is evaluated rather than the standard double integral.
+            # (Lund, R. E., and J. R. Lund, 205)
+            if (v < 2000 and not self._force_fn) or self._force_fn == "regular":
```

Comment:
@DominicChm is `self._force_fn` temporary or is it intended to be part of the public interface?

Reply:
Both `_force_fn` and `_epsabs` were meant to be temporary and used for testing, to enforce either the asymptotic or regular CDF calculation and integral precision respectively. While originally temporary, I'm wondering if it might be useful to leave them in? I can't find any existing distribution classes with instance-wide attributes like this though so it might be best to remove them?

Reply:
Also, 2000 seems to a safe max before the integral craps out around ~2500-3000.

Reply:
I think that trying to decide on the correct interface would take a lot of time / require reviewer discussion, so I think it would be best to just leave them out.

Reply:
Yeah, just document the choice.  Maybe mention it in the notes of the docstring.